src/main/java/gift/service/ProductServiceImpl.java

```diff
@@ -21,46 +22,53 @@ public ProductServiceImpl(ProductRepository productRepository) {
 
     @Override
     public ProductResponseDto createProduct(CreateProductRequestDto requestDto) {
-        Product createdProduct = productRepository.createProduct(requestDto);
-        return new ProductResponseDto(createdProduct.getId(), createdProduct.getName(),
-                createdProduct.getPrice(), createdProduct.getImageUrl());
+        Product newProduct = new Product(null, requestDto.name(), requestDto.price(),
+                requestDto.imageUrl());
+        Product savedProduct = productRepository.createProduct(newProduct);
+        return new ProductResponseDto(savedProduct.getId(), savedProduct.getName(),
+                savedProduct.getPrice(), savedProduct.getImageUrl());
```

Comment:
```suggestion
class ProductResponseDto(Product product){
 this.id = prodcut.id
 this.name = product.name
}
```

이렇게 보조 생성자를 만들어도 좋을 것 같다는 생각이 드네요 😄